### DISCUSSION TITLE
Is there a way to loop through formFields?

### DISCUSSION THREAD

**Discussion by Nilesh9768** [data-driven-forms/react-forms#1302] [Need help] [answered]
I'm trying to traverse through `formFields` in `FormTemplate` to render the fields conditionally. However, I get a blank screen when trying to traverse through `formFields` using `map/forEach`. Is there a way to achieve this? 
Hoping for a quick response. Thank you!

**Comment by rvsia** [comment, marked as answer]
@Nilesh9768 hello `formfields` is an already rendered array of React elements. I recommend to use `schema` and `useFormApi().render` instead for any more complex logic.

**Reply by Nilesh9768** [reply]
@rvsia Thanks a lot! It works for me.

**Comment by Hyperkid123** [comment]
@Nilesh9768 
> I'm trying to traverse through formFields in FormTemplate to render the fields conditionally.

I am not 100% sure what your use-case is, but did you consider using the `condition` schema config? https://data-driven-forms.org/schema/complex-condition-example#heading-complexconditionexample
